src/extension.ts

```diff
@@ -136,10 +136,18 @@ function showCoverageOnEditorChange(editor: vscode.TextEditor | undefined) {
 }
 
 function removeDecorationsOnDocumentChange(e: vscode.TextDocumentChangeEvent) {
-    // Do not remove decorations if the output document changed.
-    if (`${e.document.uri}` !== `${outputUri}`) {
-        removeDecorations();
+    // output:extension-output is the output channel for the extensions
+    // and should not be used for clearing decorations
+    if (e.document.uri.toString().startsWith('output:extension-output')) {
```

Comment:
Oh is THAT what's been clearing the decorations! That's a great find! How did you figure it out?

Reply:
I just added a log line in here in this function, but still took me a while to work out that writing to the opa output channel counted as a document!